tests/models/nougat/test_tokenization_nougat.py

```diff
@@ -0,0 +1,136 @@
+                tokenizer_r = self.rust_tokenizer_class.from_pretrained(pretrained_name, **kwargs)
+                # Simple input
+                sentence1 = "This is a simple input"
+                sentence2 = ["This is a simple input 1", "This is a simple input 2"]
+                pair1 = ("This is a simple input", "This is a pair")
+                pair2 = [
+                    ("This is a simple input 1", "This is a simple input 2"),
+                    ("This is a simple pair 1", "This is a simple pair 2"),
+                ]
+
+                # Simple input tests
+                try:
+                    tokenizer_r.encode(sentence1, max_length=max_length)
+                    tokenizer_r.encode_plus(sentence1, max_length=max_length)
+
+                    tokenizer_r.batch_encode_plus(sentence2, max_length=max_length)
+                    tokenizer_r.encode(pair1, max_length=max_length)
+                    tokenizer_r.batch_encode_plus(pair2, max_length=max_length)
+                except ValueError:
+                    self.fail("Nougat Tokenizer should be able to deal with padding")
+
+                tokenizer_r.pad_token = None  # Hotfixing padding = None
+                self.assertRaises(
+                    ValueError, tokenizer_r.encode, sentence1, max_length=max_length, padding="max_length"
+                )
+
+                # Simple input
+                self.assertRaises(
+                    ValueError, tokenizer_r.encode_plus, sentence1, max_length=max_length, padding="max_length"
+                )
+
+                # Simple input
+                self.assertRaises(
+                    ValueError,
+                    tokenizer_r.batch_encode_plus,
+                    sentence2,
+                    max_length=max_length,
+                    padding="max_length",
+                )
+
+                # Pair input
+                self.assertRaises(ValueError, tokenizer_r.encode, pair1, max_length=max_length, padding="max_length")
+
+                # Pair input
+                self.assertRaises(
+                    ValueError, tokenizer_r.encode_plus, pair1, max_length=max_length, padding="max_length"
+                )
+
+                # Pair input
+                self.assertRaises(
+                    ValueError,
+                    tokenizer_r.batch_encode_plus,
+                    pair2,
+                    max_length=max_length,
+                    padding="max_length",
+                )
+
+    @unittest.skip("NougatTokenizerFast does not have tokenizer_file in its signature")
+    def test_rust_tokenizer_signature(self):
+        pass
+
+    @unittest.skip("NougatTokenizerFast does not support pretokenized inputs")
+    def test_pretokenized_inputs(self):
+        pass
+
+    @unittest.skip("NougatTokenizerFast directly inherits from PreTrainedTokenizerFast")
+    def test_prepare_for_model(self):
+        pass
+
+
+class MarkdownCompatibleTest(unittest.TestCase):
```

Comment:
I think we need a little bit more testing here as mentioned by @amyeroberts. Would help understand the different processing done by each function. (I know it's not you but for futur reference and maintenance it will help). Just 1 test per global function is enough.

Reply:
Done thanks to @molbap